tensorflow/core/kernels/BUILD

```diff
@@ -2415,6 +2415,7 @@ cc_library(
     deps = [
         "//tensorflow/core:lib",
         "//tensorflow/core:lib_internal",
+        "//tensorflow/core:framework",
```

Comment:
The header file stack is:
```
range_sampler.h
├ random_distributions.h
└ distribution_sampler.h
  └ simple_philox.h
    └ random_distributions.h
```
So the dependency is necessary cause I modified `random_distributions.h`. Nevertheless, it can be simplified like this: https://github.com/tensorflow/tensorflow/pull/39747/files#diff-d75516433ccf13427a1719b3a290a1feL2405-L2427